### PR TITLE
feat: Compare embeddings

### PR DESCRIPTION
**Pull request by attiks**
Refs: #RW-989

### CODE REVIEW

File: html/modules/custom/reliefweb_job_tagger/src/Form/RwJobTaggerCategory.php [hunk 1]
```diff
@@ -517,4 +540,39 @@ public function getMostRelevantTermsFromEs(
     return $vocabularies;
   }
 
+  /**
+   * Get similar jobs.
+   */
+  protected function getSimilarJobs(NodeInterface $node) {
```
Comment (orakili) [inline html/modules/custom/reliefweb_job_tagger/src/Form/RwJobTaggerCategory.php:546]: This doesn't use the similarity scores from ES, is that on purpose?

File: html/modules/custom/reliefweb_job_tagger/src/Form/RwJobTaggerCategory.php [hunk 2]
```diff
@@ -517,4 +540,39 @@ public function getMostRelevantTermsFromEs(
...
+    $nid = $node->id();
+    $relevant = $this->ochaTagger->getSimilarDocuments($nid, $node->get('body')->value);
```
Comment (orakili) [inline html/modules/custom/reliefweb_job_tagger/src/Form/RwJobTaggerCategory.php:548]: Should the title be prepended to the body?

File: html/modules/custom/reliefweb_job_tagger/reliefweb_job_tagger.module [hunk 3]
```diff
@@ -340,3 +340,148 @@ function reliefweb_job_tagger_queue_job(Node $job) : void {
...
+
+  /** @var \Drupal\ocha_ai_tag\Services\OchaAiTagTagger $jobTagger */
+  $jobTagger = \Drupal::service('ocha_ai_tag.tagger');
+  $relevant = $jobTagger->getSimilarDocuments($nid, $node->get('body')->value);
+  $nodes = Node::loadMultiple($relevant);
+
+  if (isset($nodes[$nid])) {
+    unset($nodes[$nid]);
+  }
+  $categories = [];
```
[the remark this answers is not in the document]
Comment (attiks) [reply html/modules/custom/reliefweb_job_tagger/reliefweb_job_tagger.module:376]: Code in module is for testing now

File: html/modules/custom/reliefweb_job_tagger/src/Form/RwJobTaggerCategory.php (hunk 1 above)
Comment (attiks) [reply html/modules/custom/reliefweb_job_tagger/src/Form/RwJobTaggerCategory.php:546]: see https://github.com/UN-OCHA/ocha_ai/pull/47

Comment (attiks) [reply html/modules/custom/reliefweb_job_tagger/src/Form/RwJobTaggerCategory.php:555]: Switch to normalized score

File: html/modules/custom/reliefweb_job_tagger/reliefweb_job_tagger.module (hunk 3 above)
[the remark this answers is not in the document]
Comment (attiks) [reply html/modules/custom/reliefweb_job_tagger/reliefweb_job_tagger.module]: Test code removed

File: html/modules/custom/reliefweb_job_tagger/reliefweb_job_tagger.module [hunk 4]
```diff
@@ -340,3 +340,148 @@ function reliefweb_job_tagger_queue_job(Node $job) : void {
...
+  }
+
+  arsort($categories);
+  return $categories;
+}
+
+/**
+ * Index jobs.
+ */
+function reliefweb_job_tagger_index_jobs() {
```
[the remark this answers is not in the document]
Comment (attiks) [reply html/modules/custom/reliefweb_job_tagger/reliefweb_job_tagger.module]: Test code removed

File: html/modules/custom/reliefweb_job_tagger/src/Commands/ReliefJobTaggerCommands.php [hunk 5]
```diff
@@ -0,0 +1,80 @@
...
+   * @command reliefweb-jobtagger:index-jobs
+   *
+   * @usage reliefweb-jobtagger:index-jobs
+   *   Create vector index for jobs.
+   *
+   * @validate-module-enabled reliefweb_job_tagger
+   *
+   * @aliases rw-job:index
+   */
+  public function indexJobs() {
```
[the remark this answers is not in the document]
Comment (attiks) [reply html/modules/custom/reliefweb_job_tagger/src/Commands/ReliefJobTaggerCommands.php:50]: Test code removed

File: html/modules/custom/reliefweb_job_tagger/reliefweb_job_tagger.module [hunk 6]
```diff
@@ -340,3 +340,148 @@ function reliefweb_job_tagger_queue_job(Node $job) : void {
...
+
+  foreach ($job_ids as $id) {
+    reliefweb_job_tagger_index_embedding($id);
+  }
+}
+
+/**
+ * Accuracy.
+ */
+function reliefweb_job_tagger_test_accuracy() {
```
[the remark this answers is not in the document]
Comment (attiks) [reply html/modules/custom/reliefweb_job_tagger/reliefweb_job_tagger.module]: Test code removed

File: html/modules/custom/reliefweb_job_tagger/reliefweb_job_tagger.routing.yml [hunk 7]
```diff
@@ -19,3 +19,11 @@ test_ocha_ai_job_tag_theme:
     _title: 'Test job tagger - Theme'
   requirements:
     _permission: 'test ocha ai job tag'
+reliefweb_job_tagger.requeue:
+  path: '/node/{id}/re-queue'
+  defaults:
+    _title: 'Summaries comparison'
```
Comment (orakili) [inline html/modules/custom/reliefweb_job_tagger/reliefweb_job_tagger.routing.yml:25]:
```suggestion
    _title: 'Requeue job'
```

**Comment by attiks** [comment]
See https://github.com/UN-OCHA/rwint9-site/pull/851

**Pull request closed without merging**